TransferMethodRepository/Sources/TransferMethodConfigurationRepository.swift

```diff
@@ -0,0 +1,119 @@
+        [HyperwalletTransferMethodConfigurationFieldQuery: HyperwalletTransferMethodConfigurationField]()
+    private var transferMethodConfigurationKeys: HyperwalletTransferMethodConfigurationKey?
+
+    public func getFields(
+        _ country: String,
+        _ currency: String,
+        _ transferMethodType: String,
+        _ transferMethodProfileType: String,
+        completion: @escaping (Result<HyperwalletTransferMethodConfigurationField?, HyperwalletErrorType>) -> Void) {
+        let fieldsQuery = HyperwalletTransferMethodConfigurationFieldQuery(
+            country: country,
+            currency: currency,
+            transferMethodType: transferMethodType,
+            profile: transferMethodProfileType
+        )
+        guard let transferMethodConfigurationFields = transferMethodConfigurationFieldsCache[fieldsQuery] else {
+            Hyperwallet.shared
+                .retrieveTransferMethodConfigurationFields(request: fieldsQuery,
+                                                           completion: getFieldsHandler(fieldsQuery, completion))
+            return
+        }
+        completion(.success(transferMethodConfigurationFields))
+    }
+
+    public func getKeys(
+        completion: @escaping (Result<HyperwalletTransferMethodConfigurationKey?, HyperwalletErrorType>) -> Void) {
+        guard let transferMethodConfigurationKeys = transferMethodConfigurationKeys else {
+            Hyperwallet.shared
+                .retrieveTransferMethodConfigurationKeys(request: HyperwalletTransferMethodConfigurationKeysQuery(),
+                                                         completion: getKeysHandler(completion))
+            return
+        }
+        completion(.success(transferMethodConfigurationKeys))
+    }
+
+    public func refreshFields() {
+        transferMethodConfigurationFieldsCache =
+            [HyperwalletTransferMethodConfigurationFieldQuery: HyperwalletTransferMethodConfigurationField]()
+    }
+
+    public func refreshKeys() {
+        transferMethodConfigurationKeys = nil
+    }
+
+    private func getKeysHandler(
+        _ completion: @escaping (Result<HyperwalletTransferMethodConfigurationKey?, HyperwalletErrorType>) -> Void)
+        -> (HyperwalletTransferMethodConfigurationKey?, HyperwalletErrorType?) -> Void {
+        return { (result, error) in
+            self.transferMethodConfigurationKeys = CompletionHelper
+                .performCompletion(error, result, completion, self.transferMethodConfigurationKeys)
+        }
+    }
+
+    private func getFieldsHandler(
+        _ fieldQuery: HyperwalletTransferMethodConfigurationFieldQuery,
+        _ completion: @escaping (Result<HyperwalletTransferMethodConfigurationField?, HyperwalletErrorType>) -> Void)
+        -> (HyperwalletTransferMethodConfigurationField?, HyperwalletErrorType?) -> Void {
+        return { (result, error) in
+            self.transferMethodConfigurationFieldsCache[fieldQuery] = CompletionHelper
```

Comment:
I can't remove the self, because this property is inside the completion handler.  and we moved all the `guard let strongSelf = self` to be handled in the presenter.